mockafka/aiokafka/aiokafka_producer.py

```diff
@@ -28,8 +31,18 @@ class FakeAIOKafkaProducer:
     def __init__(self, *args, **kwargs) -> None:
         self.kafka = KafkaStore()
 
+    def _translate_header_to_internal_format(self, headers: Sequence[Tuple[str, bytes]]) -> dict:
+        header_dict: dict[str, bytes] = defaultdict(bytes)
+        if not headers:
+            return header_dict
+
+        for item in headers:
+            header_dict[item[0]] = item[1]
+        return header_dict
+
     async def _produce(self, topic, value=None, *args, **kwargs) -> None:
         # create a message and call produce kafka
+        kwargs['headers'] = self._translate_header_to_internal_format(kwargs['headers'])
```

Comment:
As `headers` is now a required key I'd suggest making it an explicitly named argument (still keyword only), even if the result is that it's assigned into `kwargs` for passing onwards. This allows us to both clarify to callers about its nature and specify a default if needed.